### ISSUE TITLE
Operations on multiple VMs

### ISSUE THREAD

**Issue by ahmetalpbalkan** [Azure/azure-xplat-cli#1684]
Operations like "azure vm delete" or start/stop/reboot etc. works with only one parameter. 

This is inconvenient in the linux world. Most people do scripting like

    azure vm list | (..get names..) |  grep ^name | xargs azure vm delete

Similarly other cloud platforms' cli tools easily allow this and it helps a great deal in scripting without for loops etc. 

Another advantage is, this command can handle parallelization and waiting on results, which is mot really an easy thing to do, especially on windows command line (cmd.exe).

**Comment by devigned** [comment]
:+1:

**Comment by bureado** [comment]
+1, operations that block the cloud service (i.e., I'm deleting a VM from a cloud service and can't delete the others while that operation is in progress) are really annoying and some sort of queue management in `azure-cli` (or in Azure side, FWIW) would be really useful (here's a use case: piping `azure vm create -c` thru `parallel -j n`)

Re. scripting `azure-cli` output what is done today is adding `--json` and using `jq` or `jsawk`, I believe this is covered in documentation today.

**Comment by ahmetalpbalkan** [comment]
@bureado jq/jsawk is what i refer by (..get names..) above. 

Current service management REST API has some concurrency limitations if VMs are under the same affinity group etc but retries or queueing is doable

In any case ARM methods should hve this functionality because there's no such constraints there.

**Comment by squillace** [comment]
+1. I was just going to file another issue on this, because **some** operations have a --no-wait switch, and I was going to ask for one universally. When is this coming out....?

**Comment by ahmetalpbalkan** [comment]
ETOOADVANCEDFORUS
